src/stores/RouterStore.ts

```diff
@@ -223,6 +233,15 @@ export class RouterStore {
             }
         }
 
+        // --- globalBeforeEnter ---
```

Comment:
This code will have to move to TransistionState.ts, where the other hooks are. Which means that it will need access the option like this: this.routerStore.options.globalBeforeEnter